File: node/crc32.js
/*jshint undef:true node:true strict:false*/
var fs = require('fs');

var Crc32 = function() {
  this.table = [];
  this.value = null;

  var v;

  for(var i = 0; i < 256; ++i) {
    v = i;
    for(var j = 0; j < 8; ++j) {
      v = (v & 1) ? (0xedb88320 ^ (v >>> 1)) : (v >>> 1);
    }
    this.table[i] = v | 0;
  }
};

Crc32.prototype.start = function() {
  this.value = 0xffffffff;
};

Crc32.prototype.update = function(text) {
  for(var i = 0; i < text.length; ++i) {
    var c = text.charCodeAt(i);
    this.value = this.table[(this.value ^ c) & 0xFF] ^ (this.value >>> 8);
  }
};

Crc32.prototype.finalize = function() {
  return (this.value ^ 0xffffffff) >>> 0;
};

for(var i = 2; i < process.argv.length; ++i) {

  (function(filename) {
    var crc    = new Crc32();
    var stream = fs.createReadStream(filename, {
      encoding:   'binary',
      flags:      'r',
      bufferSize: 1024 * 1024
    }).addListener('data', function(chunk) {
      crc.update(chunk);
    }).addListener('open', function() {
      crc.start();
    }).addListener('close', function() {
      console.log(filename, crc.finalize().toString(16));
    });
  })(process.argv[i]);

}

File: go/crc32.go
package main

import (
	"os"
	"fmt"
)

type Crc32 struct {
	value uint32
	table[256] uint32
}

func (self *Crc32) Init() {
	var v uint32

	for i := 0; i < 256; i++ {
		v = uint32(i)
		for j := 0; j < 8; j++ {
			if (v & 1 == 1) {
				v = (0xedb88320 ^ (v >> 1))
			} else {
				v >>= 1
			}
		}
		self.table[i] = v
	}
}

func (self *Crc32) Start() {
	self.value = 0xffffffff
}

func (self *Crc32) Finalize() (uint32) {
	return self.value ^ 0xffffffff
}

func (self *Crc32) Update(data []byte) {
	for i := 0; i < len(data); i++ {
		self.value = self.table[(self.value ^ uint32(data[i])) & 0xFF] ^ (self.value >> 8)
	}
}

func main() {
	var args = os.Args[1:]
	var crc  = &Crc32{}

	crc.Init()

	for i := 0; i < len(args); i++ {
		fp, err := os.Open(args[i])
		if (err == nil) {
			crc.Start()
			data := make([]byte, 1024 * 1024)

			for {
				count, err := fp.Read(data)
				if (err != nil || count == 0) {
					break
				}
				crc.Update(data[:count])
			}
			fmt.Printf("%s: %08X\n", args[i], crc.Finalize())
			fp.Close()
		}
	}

}